tools/cli/commands.go

```diff
@@ -1041,6 +1090,54 @@ func listClosedWorkflow(client client.Client, pageSize int, earliestTime, latest
 	return response.Executions, response.NextPageToken
 }
 
+func getListResultInRaw(c *cli.Context, queryOpen bool, nextPageToken []byte) ([]*s.WorkflowExecutionInfo, []byte) {
+	wfClient := getWorkflowClient(c)
+
+	earliestTime := parseTime(c.String(FlagEarliestTime), 0)
+	latestTime := parseTime(c.String(FlagLatestTime), time.Now().UnixNano())
+	workflowID := c.String(FlagWorkflowID)
+	workflowType := c.String(FlagWorkflowType)
+	pageSize := c.Int(FlagPageSize)
+	if pageSize <= 0 {
+		pageSize = defaultPageSizeForList
+	}
+
+	var workflowStatus s.WorkflowExecutionCloseStatus
+	if c.IsSet(FlagWorkflowStatus) {
+		if queryOpen {
+			ErrorAndExit(optionErr, errors.New("you can only filter on status for closed workflow, not open workflow"))
+		}
+		workflowStatus = getWorkflowStatus(c.String(FlagWorkflowStatus))
+	} else {
+		workflowStatus = workflowStatusNotSet
+	}
+
+	if len(workflowID) > 0 && len(workflowType) > 0 {
+		ErrorAndExit(optionErr, errors.New("you can filter on workflow_id or workflow_type, but not on both"))
+	}
+
+	var result []*s.WorkflowExecutionInfo
+	if queryOpen {
+		result, nextPageToken = listOpenWorkflow(wfClient, pageSize, earliestTime, latestTime, workflowID, workflowType, nextPageToken, c)
+	} else {
+		result, nextPageToken = listClosedWorkflow(wfClient, pageSize, earliestTime, latestTime, workflowID, workflowType, workflowStatus, nextPageToken, c)
+	}
+
+	return result, nextPageToken
+}
+
+// default will print decoded raw
+func printListResults(executions []*s.WorkflowExecutionInfo, inJson bool) {
+	for _, execution := range executions {
+		if inJson {
+			j, _ := json.Marshal(execution)
+			fmt.Println(string(j))
```

Comment:
Will it produce the parsable JSON list of execution structures?

Reply:
I see your point. It currently produces json lines,  
updated to produce a JSON list (also update top comments with latest result)